tests/004-streams.cpp

```diff
@@ -153,24 +153,31 @@ namespace oxen::quic::test
 
         std::atomic<int> index{0};
         std::atomic<int> data_check{0};
-        opt::max_streams max_streams{8};
-        std::vector<std::shared_ptr<Stream>> streams{12};
+        int n_threads = (std::thread::hardware_concurrency() < 12) ? std::thread::hardware_concurrency() : 12;
+
+        if (n_threads < 8)
+            SKIP("Current hardware does not support ad hoc creation of > 8 threads!");
```

Comment:
This isn't necessary; you can still create however many threads you want, even if you don't have the CPUs to back them (they just got scheduled across the cores you do have).